pkg/services/ngalert/state/manager.go

```diff
@@ -375,9 +380,14 @@ func (st *Manager) saveAlertStates(ctx context.Context, logger log.Logger, state
 		err = st.instanceStore.SaveAlertInstance(ctx, instance)
 		if err != nil {
 			logger.Error("Failed to save alert state", "labels", s.Labels.String(), "state", s.State, "error", err)
+			return nil
 		}
+		return nil
 	}
-	logger.Debug("Saving alert states done", "count", len(states))
+
+	logger.Debug("Saving alert states", "count", len(states), "max_concurrent_state_savers", st.maxConcurrentStateSavers)
+	_ = concurrency.ForEachJob(ctx, len(states), st.maxConcurrentStateSavers, saveState)
```

Comment:
Though the `ForEachJob` function will work as expected with concurrency=1, it will spawn a Goroutine needlessly. This is probably not anything to be worried about, just flagging it.